### PR TITLE
Add support for CreditNote

### PR DESCRIPTION
**Pull request by remi-stripe**
cc @stripe/api-libraries

### CODE REVIEW

File: src/Stripe.net/Entities/CreditNote.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        /// <summary>
+        /// The link to download the PDF of the credit note.
+        /// </summary>
+        [JsonProperty("pdf")]
+        public string Pdf { get; set; }
+
+        /// <summary>
+        /// Reason for issuing this credit note, one of <code>duplicate</code>,
+        /// <code>fraudulent</code>, <code>order_change</code>, or
+        /// <code>product_unsatisfactory</code>.
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/CreditNote.cs]: `<code>` -> `<c>`

File: src/Stripe.net/Entities/CreditNote.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            set
+            {
+                StringOrObject<Invoice>.Map(value, s => this.InvoiceId = s, o => this.Invoice = o);
+            }
+        }
+        #endregion
+
+        /// <summary>
+        /// Has the value <code>true</code> if the object exists in live mode or the value
+        /// <code>false</code> if the object exists in test mode.
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/CreditNote.cs]: `<code>` -> `<c>`

File: src/Stripe.net/Services/CreditNotes/CreditNoteCreateOptions.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        [JsonProperty("refund_amount")]
+        public long? RefundAmount { get; set; }
+
+        /// <summary>
+        /// Reason for issuing this credit note, one of <code>duplicate</code>,
+        /// <code>fraudulent</code>, <code>order_change</code>, or
+        /// <code>product_unsatisfactory</code>.
+        /// </summary>
+        [JsonProperty("reason")]
+        public string Reason { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Services/CreditNotes/CreditNoteCreateOptions.cs]: Should be before `refund`.

File: src/Stripe.net/Services/CreditNotes/CreditNoteCreateOptions.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        /// Amount to refund. If set, a refund will be created for the charge associated with the
+        /// invoice.
+        /// </summary>
+        [JsonProperty("refund_amount")]
+        public long? RefundAmount { get; set; }
+
+        /// <summary>
+        /// Reason for issuing this credit note, one of <code>duplicate</code>,
+        /// <code>fraudulent</code>, <code>order_change</code>, or
+        /// <code>product_unsatisfactory</code>.
```
Comment (ob-stripe) [inline src/Stripe.net/Services/CreditNotes/CreditNoteCreateOptions.cs]: `<code>` -> `<c>`

**Comment by remi-stripe** [comment]
@ob-stripe Thanks for the review. I made some changes (and added 2 missing fields). I did not force push to make the re-review easier (and I'll squash after approval).

Can you re-r? the last 2 commits?

**Comment by remi-stripe** [comment]
> LGTM, though the new invoice fields are not in the public spec yet.

Yep those are tricky to align unfortunately, I originally missed them

**Pull request merged**